### PR TITLE
[SYCL] Added regression test for non uniform work group size.

### PR DESCRIPTION
**Pull request by etdamiba-intel**
Signed-off-by: Eugene T Damiba <eugene.t.damiba@intel.com>

### CODE REVIEW

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: opencl
+// REQUIRES: level_zero
+//==------------------- non-uniform-wk-gp-test.cpp
+//--------------------------==//
```
Comment (alexbatashev) [inline sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:6]:
```suggestion
//==------------------- non-uniform-wk-gp-test.cpp --------------------------==//
```

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            [=](nd_item<3> itm) {
+              int i = itm.get_global_id(0);
+              int j = itm.get_global_id(1);
+              int k = itm.get_global_id(2);
+            });
+      });
+    }
+  }
+
+  catch (runtime_error &E) {
```
Comment (alexbatashev) [inline sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:37]:
```suggestion
  } catch (sycl::runtime_error &E) {
```

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  for (const auto &plt : platform::get_platforms()) {
+    if (!plt.has(aspect::host)) {
+      std::cout << "Platform #" << pltCount++ << ":" << std::endl;
+      if (plt.get_backend() == backend::level_zero) {
+        std::cout << "Backend: Level Zero" << std::endl;
+        ret = test();
+      }
+    }
+    std::cout << std::endl;
+  }
```
Comment (alexbatashev) [inline sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:68]: Why do you need this code? LIT shouldn't run your test on anything but L0 GPU.

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    const int N = 1;
+    {
+      q.submit([&](handler &cgh) {
+        cgh.parallel_for<class test_kernel>(
+            nd_range<3>(range<3>{1, 1, N}, range<3>{1, 1, 16}),
+            [=](nd_item<3> itm) {
+              int i = itm.get_global_id(0);
+              int j = itm.get_global_id(1);
+              int k = itm.get_global_id(2);
```
Comment (alexbatashev) [inline sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:31]: `i`, `j`, and `k` are not used. Doesn't compiler yield a warning here?

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+int test() {
+  try {
+    queue q = queue();
+    auto device = q.get_device();
+    auto deviceName = device.get_info<cl::sycl::info::device::name>();
+    std::cout << " Device Name: " << deviceName << std::endl;
+
+    const int N = 1;
+    {
```
Comment (alexbatashev) [inline sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:24]: These brackets are meaningless.

**Review by alexbatashev — CHANGES_REQUESTED** [review]
Overall LGTM, please, fix the formatting in the file.

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: opencl
+// REQUIRES: level_zero
```
Comment (smaslov-intel) [inline sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:4]: why are other backends filtered?

Comment (alexbatashev) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:4]: AFAIK, OpenCL 2.1 allows non uniform work-group sizes.

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp (hunk 3 above)
Comment (etdamiba-intel) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:67]: LIT test should be able to compiled and run manually as well.

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp (hunk 4 above)
Comment (etdamiba-intel) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:30]: Replaced with a new code.

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp (hunk 6 above)
Comment (etdamiba-intel) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:4]: This is a negative test to catch a specific error message happening with the level zero backend only.

**Review by alexbatashev — APPROVED** [review]
LGTM

File: sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp (hunk 6 above)
Comment (smaslov-intel) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:4]: Where is the specific message being checked?

Comment (etdamiba-intel) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:4]: The message is on line 35:
if (std::string(E.what()).find("Specified local size doesn't match the required work-group size specified in the program source") != std::string::npos)

Comment (smaslov-intel) [reply sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp:4]: indeed, thanks

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**